### ISSUE TITLE
States: UK place names are not specific enough

### ISSUE THREAD

**Issue by djjudas21** [photoprism/photoprism#2097]
### Discussed in https://github.com/photoprism/photoprism/discussions/2095

<div type='discussions-op-text'>

<sup>Originally posted by **djjudas21** March  2, 2022</sup>
Hi there. I'm a new PhotoPrism user and I'm checking out the functionality on some holiday photos I took last month to see if it meets my needs before I migrate my whole library.

I noticed that the place names are not consistent. Where I live, in the UK, the location appears in the "States" view simply as England. Whereas the photos I took on holiday in Switzerland seem to appear either as the state (Basel-Landschaft) or the city (Zurich), as I would expect.

![Screenshot from 2022-03-02 17-40-50](https://user-images.githubusercontent.com/604595/156417794-8ae7b2fc-0f57-4734-b261-8b65132cdda3.png)

I don't know how your location data is formatted, but would it be possible for you to tweak your backend when working with UK locations? For example, I would classify my current location as:

> Country: UK
> Subdivision: England
> State: Bristol
> City: Bristol

and another location I visit frequently:

> Country: UK
> Subdivision: England
> State: Warwickshire
> City: Nuneaton

Here, I don't really care about the "subdivision" part of the location (England, Scotland, Wales, Northern Ireland) and the "state" or even the "city" part of the location is much more useful. I think most UK residents would probably agree with me.

I guess the UK is different from most other countries here, whose classification would go straight from "country" to "state" without the subdivision.

Thanks :smile: </div>

**Comment by aaronrudkin** [comment]
I noticed a similar issue with **Ireland** that I'll post here rather than create another issue; this is not high priority but I figure whenever you get time to update the States for the UK you can do the same for Ireland.

Ireland has two major sub-national divisions:
- Historical provinces (of which there are four: Leinster, Munster, Connacht, and Ulster)
- Counties (of which there are 26)

Separately, the county of Northern Ireland occupies some of the north of the island of Ireland. This country is a part of the United Kingdom and is not a part of Ireland. I would say most Irish would tell you the county is the most significant subnational level, and thus the appropriate equivalent to a "State" here. But provinces could also be an option. Either is fine.

Currently, all photos in Leinster, Munster, and Connacht do not show up in states, either with their province or their county attached.  Photos in the counties which make up Ulster but are part of Ireland are listed as "Northern Ireland, Ireland". This is absolutely not correct. Photos in Northern Ireland correctly list themselves as Northern Ireland, United Kingdom. 

This is something of a political issue as there are political factions in both Ireland and Northern Ireland that favor reunification and factions that oppose reunification and this particular tension has caused periodic violence for the last 60+ years. But it's clear that nothing named "Northern Ireland" is currently legally part of Ireland.

Some example data:

- County Donnegal, part of Ulster province, Ireland: 54.47746 lat, -8.281295 long (this will resolve as Northern Ireland, Ireland, which is INCORRECT)

- Belfast, a town in Country Antrim, part of Northern Ireland, United Kingdom: 54.599274 lat, -5.9513445 long (this will resolve as Northern Ireland, United Kingdom, which is correct)

- Coney Island, part of County Sligo, which is part of Connacht province, Ireland: 53.325417 lat, -7.9884334 long (this will not resolve as anything in "states")


Preferred solution:
- List the counties of Ireland as states within Ireland on the states pge.

Second best solution:
- List the provinces of Ireland as states within Ireland

Third best solution:
- Do not tag images taken in the counties of Ulster province, Ireland as "Northern Ireland, Ireland".

**Comment by ilazgo** [comment]
**TL;DR:** In Spain the term "State" refers to Spain. We should look for a generic name for the subdivisions, or we should look for another structure per country.

Something similar happens in Spain. The naming could be a bit political here.

What you name "State" is "Comunidad Autónoma" (Autonomous Community), and what you name "Country" is "Estado" (State). And we have more divisions than it makes sense to make an album, like "Provincias" (Provinces). So, it's a bit complex :).

For example, in Spain we have:

- State: Spain
- Autonomous community: Galicia
- Province: Pontevedra

But the possible problem here is that what is being called "State" (translated as "Estado") is what in Spain is a autonomous community. And here in Spain we have several independence movements (such as Galicia, the Basque Country or Catalonia), so this could be a political issue.

We cannot change the translation and translate "State" into Spanish as "Comunidad Autónoma" because "Comunidad Autónoma" is only valid for Spain, in other countries, even Spanish-speaking, the second division is called something else.

Maybe we should look for some generic name for the subdivisions (_Region?_). Or we should have another structure with countries and their divisions. For example:

- USA
  - States
- Spain
  - Autonomous communities
- Ireland
  - Historical provinces
  - Counties
